### PR TITLE
Feature/pas 12528   2023 08 29 1016   wp plugin causes 404 issues in wp content after activation syncing

### PR DESCRIPTION
**Pull request by klikas**

### CODE REVIEW

File: class/PostTypes/CptBase.php [hunk 1]
```diff
@@ -74,6 +74,8 @@ public static function create_rewrite_rules()
     $template_variable = $resource->get_permalink_template_variable();
     $post_permalink_template = static::get_permalink_template();
 
+    if(!$post_permalink_template) return;
```
Comment (felixpackard) [inline class/PostTypes/CptBase.php:77]: This is a base class used for posts and people FYI - but there's no need for this as the post and person permalink templates aren't allowed to be empty.

File: class/Services/RewriteService.php [hunk 2]
```diff
@@ -29,8 +31,6 @@ public static function add_preview_rewrite()
 
     static::add_rewrite_rule($regex, $query);
     static::add_rewrite_tag('%passle_preview%');
-
-    flush_rewrite_rules();
```
Comment (felixpackard) [inline class/Services/RewriteService.php]: I think we actually do want this here. The issue with the old code is it was adding an action to the init event then flushing the rewrite rules, so it was being flushed before we'd added our new rule. Having it here is correct.

File: class/Services/OptionsService.php [hunk 3]
```diff
@@ -34,6 +34,8 @@ public static function set(Options $options)
     }
 
     RewriteService::add_preview_rewrite();
+
+    flush_rewrite_rules();
```
Comment (felixpackard) [inline class/Services/OptionsService.php:38]: We can remove this from here.

**Review by felixpackard — CHANGES_REQUESTED** [review]

File: class/PostTypes/CptBase.php (hunk 1 above)
Comment (klikas) [reply class/PostTypes/CptBase.php]: I have now removed this check

**Review by felixpackard — APPROVED** [review]

**Pull request merged**